modules/gapi/samples/pipeline_modeling_tool/pipeline_builder.hpp

```diff
@@ -439,8 +533,36 @@ Pipeline::Ptr PipelineBuilder::construct() {
             // (3). Extract proto input from every input node.
             inputs.push_back(in_data.arg.value());
         }
+        // NB: If node shouldn't be called on each iterations,
+        // it should be wrapped into subgraph which is able to skip calling.
+        if (call.call_every_nth != 1u) {
+            // FIXME: Limitation of the subgraph operation (<GMat(GMat)>).
+            // G-API doesn't support dynamic number of inputs/outputs.
+            if (inputs.size() > 1u) {
+                throw std::logic_error(
+                        "skip_frame_nth is supported only for single input subgraphs");
+            }
+
+            if (outputs.size() > 1u) {
+                throw std::logic_error(
+                        "skip_frame_nth is supported only for single output subgraphs");
+            }
+            // FIXME: Should be generalized.
+            // Now every subgraph contains only single node
+            // which has single input/output.
+            GAPI_Assert(cv::util::holds_alternative<cv::GMat>(inputs[0]));
+            cv::GProtoArgs subgr_inputs{cv::GProtoArg{cv::GMat()}};
+            cv::GProtoArgs subgr_outputs;
+            call.run(subgr_inputs, subgr_outputs);
+            auto comp = cv::GComputation(cv::GProtoInputArgs{subgr_inputs},
+                                         cv::GProtoOutputArgs{subgr_outputs});
+            call = CallNode{call.name,
```

Comment:
do we over-wrap existing call operation with call_every_nth  != 1 by a new operation with call_every_nth == 1 which has a counter and the same comp?
If my understanding is correct than in this case do we just return the previous Mat without actual calculation? if yes then why it is called as "skip frames"? I thought that by skip frames it means receiving something empty frames of increasing next frame awaiting duration

Reply:
Agree, naming is confusing. This is about calling some nodes(kernels) on every `n` iterations. (not on every iteration)

Reply:
I see,
Then question - Does it affect final measurement is some sort of way? maybe it had better to print information about node invocation count also in final report?

Reply:
If node works not on every frame, it obviously improves performance.
> maybe it had better to print information about node invocation count also in final report


Not sure that this is useful info, let's wait for the feedback firstly